dev-util/radare2/radare2-5.9.4.ebuild

```
@@ -0,0 +1,123 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit bash-completion-r1 toolchain-funcs
+
+DESCRIPTION="unix-like reverse engineering framework and commandline tools"
+HOMEPAGE="https://www.radare.org"
+
+ARM64_COMMIT=55d73c6bbb94448a5c615933179e73ac618cf876
+ARMV7_COMMIT=f270a6cc99644cb8e76055b6fa632b25abd26024
+BINS_COMMIT=bca0e277a234cc71b43d919f88bda03cd4a518fd
+
+if [[ ${PV} == *9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/radareorg/radare2"
+else
+	SRC_URI="mirror+https://github.com/radareorg/radare2/archive/${PV}.tar.gz -> ${P}.tar.gz
+		mirror+https://github.com/radareorg/vector35-arch-arm64/archive/${ARM64_COMMIT}.tar.gz -> ${P}-vector35-arm64.tar.gz
+		mirror+https://github.com/radareorg/vector35-arch-armv7/archive/${ARMV7_COMMIT}.tar.gz -> ${P}-vector35-armv7.tar.gz
+		test? ( https://github.com/radareorg/radare2-testbins/archive/${BINS_COMMIT}.tar.gz -> radare2-testbins-${BINS_COMMIT}.tar.gz )
+	"
+
+	KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+fi
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="ssl test"
+
+# Need to audit licenses of the binaries used for testing
+RESTRICT="fetch !test? ( test )"
+
+RDEPEND="
+	app-arch/lz4
+	>=dev-libs/capstone-5.0_rc4:=
+	dev-libs/libzip:=
+	dev-libs/xxhash
+	sys-apps/file
+	sys-libs/zlib
+	ssl? ( dev-libs/openssl:0= )
+"
+DEPEND="
+	${RDEPEND}
+	dev-util/gperf
+"
+BDEPEND="virtual/pkgconfig"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-5.8.2-vector35.patch"
+	"${FILESDIR}/${PN}-5.8.4-test.patch"
+)
+
+src_prepare() {
+	default
+
+	# Hack for vector35 arm plugins
+	mv "${WORKDIR}/vector35-arch-arm64-${ARM64_COMMIT}" libr/arch/p/arm/v35/arch-arm64 || die
+	mv "${WORKDIR}/vector35-arch-armv7-${ARMV7_COMMIT}" libr/arch/p/arm/v35/arch-armv7 || die
+
+	if use test; then
+		cp -r "${WORKDIR}/radare2-testbins-${BINS_COMMIT}" "${S}/test/bins" || die
+		cp -r "${WORKDIR}/radare2-testbins-${BINS_COMMIT}" "${S}" || die
+	fi
+
+	# Fix hardcoded docdir for fortunes
+	sed -i -e "/^#define R2_FORTUNES/s/radare2/$PF/" \
+		libr/include/r_userconf.h.acr || die
+}
+
+src_configure() {
+	# Ideally these should be set by ./configure
+	tc-export CC AR LD OBJCOPY RANLIB
+	export HOST_CC=${CC}
+
+	econf \
+		--with-syscapstone \
+		--with-syslz4
+		--with-sysmagic \
```

```suggestion
		--with-syslz4 \
		--with-sysmagic \
```
Although we'd prefer an array form due to this:
```
^[[33m * ^[[39;49;00mQA Notice: command not found:
^[[33m * ^[[39;49;00m
^[[33m * ^[[39;49;00m   /var/tmp/portage/dev-util/radare2-5.9.4/temp/environment: line 1288: --with-sysmagic: command not found
```